CONTRIBUTING.md

```diff
@@ -0,0 +1,89 @@
+6. Install all required dependencies with `npm install`
+7. Then run `npm run all` to set up io-site.
+8. Finally, run `npm run dev` to start the server on port 5173.
+
+Congrats! 🎉 You should now be able to use io-site by navigating to [http://127.0.0.1:5173/](http://localhost:5173/) on Chrome. It is also possible to open a new window with `o + enter` via the terminal the io-site is running in.
+
+### Setting up the VSCode debugger
+VSCode provides a debugging mode using Chrome. To use it, follow these steps:
+
+1. Go to **Run>Add Configuration** and select `{} Chrome: Launch`
+2. This will create `launch.json`.
+3. Make sure to add the `/site/` path to `webRoot` and change `localhost` to `127.0.0.1` on `url`, so that the file looks like this:
+```json
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "type": "chrome",
+            "request": "launch",
+            "name": "Launch Chrome against localhost",
+            "url": "http://127.0.0.1:5173",
+            "webRoot": "${workspaceFolder}/site/"
+        }
+    ]
+}
+```
+
+Great! You should now be able to run the io-site server with `npm run dev` and then the debugger by either pressing **F5** on your keyboard, or clicking on **Run>Start Debugging**.
+
+This will launch Chrome on whatever address you provided to `url` on `launch.json`, allowing you to use io-site as well as entering debugging mode on VSCode.
+
+> **Note:** Again, you are welcome to try your own debugging method, such as [Chrome DevTools](https://developer.chrome.com/docs/devtools/javascript/breakpoints/).
+
+
+## Directory structure
+
+Now that you know how to set up, run and debug io-site, you will probably want a tour of the directory structure. This should help you know where to add any new features, or where to spot a particular bug in need of fixing.
+
+Here are the relevant parts of the directory tree along with short descriptions of each directory:
+
+- `.github/workflows`: Contains the build and deploy workflows for the project.
+- `site`: The 'site' directory is the heart of the codebase, housing the majority of substantial code. Subdirectories include:
+    - `public`: This contains shared images and vector files that are shared sitewide.
+    - `scripts`: The home for general purpose scripts, such as 'deploy'.
+    - `src`: Webpage code is stored here. Currently, styling and functional code are stored in the same directory, using the practice of `FILENAME.jsx` and `FILENAME.css`. Subdirectories include:
+        - `nav`: Contains the navigational components of the io-site which can be seen at the top of the web page. Similarly, styling and functionality files are kept in the same location, with only file endings differing in naming.
```

Comment:
My preference is to avoid this level of detail in the base README. This almost always will get out of date. A README in the base of nav will be much more likely to stay up to date.

Reply:
Do you mean at the 'nav' level only or the rest of the directory summaries as well?